### PR TITLE
Improved email input for group members and collaborators

### LINKED ISSUE #131: Empty group member list gives success message
Adding group members without any emails shows the success message that a groups member has been added. It should show an error message regarding the empty email field. 
Similar issue occurs while adding collaborators.

### PR DESCRIPTION
**Pull request by ankitkataria**
Fixes #131 

Now the message is of the format:
`Out of 3 Email(s), 2 were valid and 1 was invalid. 1 user(s) will be invited. 1 user(s) was already present.` 
 or `Out of 4 Email(s), 2 were valid and 2 were invalid. 2 user(s) will be invited. No users were already present.`  or `No valid Email(s) entered.`

### CODE REVIEW

File: app/controllers/collaborations_controller.rb [hunk 1]
```diff
@@ -35,20 +35,23 @@ def create
       render plain: "Access Restricted " and return
     end
 
-    collaboration_emails = collaboration_params[:emails].split(/[\s,\,]/).compact.reject(&:empty?)
+    already_present = User.where(Collaboration.where(project_id: @project.id).select(:user_id)).select(:email).map(&:email)
```
Comment (tachyons) [inline app/controllers/collaborations_controller.rb]: You can use 'pluck' here

Comment (ankitkataria) [reply app/controllers/collaborations_controller.rb]: Done :heavy_check_mark:

File: app/controllers/collaborations_controller.rb [hunk 2]
```diff
@@ -35,20 +35,23 @@ def create
       render plain: "Access Restricted " and return
     end
 
-    collaboration_emails = collaboration_params[:emails].split(/[\s,\,]/).compact.reject(&:empty?)
+    already_present = User.where(Collaboration.where(project_id: @project.id).select(:user_id)).pluck(:email)
```
Comment (tachyons) [inline app/controllers/collaborations_controller.rb]:
```ruby
User.where(@project.collborations.pluck(:user_id)).pluck(:email)
```

File: app/controllers/collaborations_controller.rb [hunk 3]
```diff
@@ -35,20 +35,23 @@ def create
...
+    newly_added.each do |email|
       email = email.strip
       user = User.find_by(email: email)
       if user.nil?
         # PendingInvitation.where(group_id:@group.id,email:email).first_or_create
       else
         Collaboration.where(project_id:@project.id,user_id:user.id).first_or_create
       end
-    end
+    end if newly_added.present?
```
Comment (tachyons) [inline app/controllers/collaborations_controller.rb]: needed?

**Comment by ankitkataria** [comment]
Thinking about it, no.

File: app/controllers/collaborations_controller.rb (hunk 3 above)
Comment (ankitkataria) [reply app/controllers/collaborations_controller.rb]: No, I removed it.

**Comment by ankitkataria** [comment]
@tachyons If there is anything more left to be done in the PR, let me know. :smile:

**Comment by ankitkataria** [comment]
@satu0king are there any improvements required here for the message?

**Comment by armudgal** [comment]
Hi, can we squash all the commits into one. Thanks

**Comment by ankitkataria** [comment]
Done :heavy_check_mark:

**Pull request merged**